lib/decorateWithMetrics.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Each key is a method name and the value is an array of calls metadata
+ * @type {Map<String, Array<CallMeta>>}
+ */
+const stats = new Map();
+
+/**
+ * Class instances are easily tracked by Dev tools memory snapshots
+ * Wrapping a result with such instance allows us to track it without
+ * incurring extra weight
+ */
+class TaggedItem {
+    constructor(data) {
+        this.ref = data;
+    }
+}
```

Comment:
So far this seems to only work on web - when I capture profile samples in react-native / debug / chrome / memory - the objects do not appear there, while in web / chrome / memory I see this 

![image](https://user-images.githubusercontent.com/12156624/118067697-bd32cc80-b3a9-11eb-8f36-ec49a7c184a5.png)

Reply:
I've posted more info here (2nd part): https://github.com/Expensify/react-native-onyx/pull/70#pullrequestreview-660153513
If you decide to keep this functionality I can add more details in the `README`